app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -732,6 +735,11 @@ class HomeFragment : Fragment() {
             requireComponents.core.tabCollectionStorage.getCollections().observe(this, observer)
         }
     }
+    private fun singleUseCollectionsTelemetryObserver() {
+        requireComponents.core.tabCollectionStorage.getCollections().observeOnce {
+            Collections.totalCollections.add(it.size)
```

Comment:
Since the metrics ping is submitted once a day and this can be called more than once a day, I'm concerned that we'll add together the current collections count multiple times a day, e.g. if the app is started at 5pm and 6pm and the user has 5 collections, we'd get 5 + 5 = 10 instead of 5.

I think we either need to:
- submit the ping each time this metric is collected (in which case, maybe it should be a part of `startup-timeline` because it's transmitted once per start/stop cycle; though I'm not sure what happens if the application crashes - maybe this strategy won't work then)
- add code to ensure we only call this once per ping lifecycle (e.g. once a day for metrics) though this seems a bit tricky too...

@Dexterp37 Is that right or am I misunderstanding how counters work?

Reply:
I set the life cycle of the metric to be `application`. With this, I believe the metric gets reset every time the app restarts. For example, if we start the app at 5 pm with 6 collections and then again at 6 pm with 5 collections. The metric will only report 5 (which is the latest). Unless I missunderstood the usecase for lifecycle @Dexterp37 ?

Reply:
> 
> 
> Since the metrics ping is submitted once a day and this can be called more than once a day, 

Does this get called multiple times during a single Fenix run?

> I'm concerned that we'll add together the current collections count multiple times a day, e.g. if the app is started at 5pm and 6pm and the user has 5 collections, we'd get 5 + 5 = 10 instead of 5.

@sraturi set the correct `lifetime` for this, meaning that next time Fenix starts the metric will be cleared. So, if this function gets called multiple time during a Fenix run, then your concern is valid. Otherwise, if it only runs once, you should not be worried :)

> I think we either need to:
> 
>     * submit the ping each time this metric is collected (in which case, maybe it should be a part of `startup-timeline` because it's transmitted once per start/stop cycle; though I'm not sure what happens if the application crashes - maybe this strategy won't work then)
> 
>     * add code to ensure we only call this once per ping lifecycle (e.g. once a day for metrics) though this seems a bit tricky too...
> 
> 
> @Dexterp37 Is that right or am I misunderstanding how counters work?

In general, you should not really care about _when_ built-in pings are sent. This is opaque for this exact reason: the ping scheduler dictates when the "snapshot" of the current values is taken. Governing when things are cleared is done through [lifetimes](https://mozilla.github.io/glean/book/user/metric-parameters.html#optional-metric-parameters).

> 
> 
> I set the `lifetime` of the metric to be `application`. With this, I believe the counter gets reset every time the app restarts. For example, if we start the app at 5 pm with 6 collections and then again at 6 pm with 5 collections. The metric will only report 5 (which is the latest). Unless I misunderstood the use case for lifecycle @Dexterp37 ?

Yes, that's true @sraturi :)